### DISCUSSION TITLE
Basic colors manipulation

### DISCUSSION THREAD

**Discussion by Solido** [btwld/mix#469] [Q&A] [answered]
I would like to discuss basic color manipulation so instead of 
declaring all colors upfront.

``` dart
Style(
  $box.color.ref(refcolor),
  $state1 (
    $box.color.ref(state1refcolor)
  ),
  $state2 (
    $box.color.ref(state2refcolor)
  )
)
```

We can build around only 1 ref color.

``` dart
Style(
  $box.color.ref(refcolor),
  $state1 (
    $box.color.ref(refcolor).brighten(10)
  ),
  $state2 (
    $box.color.ref(refcolor).darken(10)
  )
)
```

Combined with style merging it would greatly reduce the length of declarations
only requiring base color.

Thanks for your feedback!

**Comment by tilucasoli** [comment, marked as answer]
The following code can be used to achieve the same result.

```dart
Style(
  $box.color.ref(refcolor),
  $state1 (
    $box.color.brighten(10)
  ),
  $state2 (
    $box.color.darken(10)
  )
)
```

However, I can understand why you would want color manipulation in the same line.
